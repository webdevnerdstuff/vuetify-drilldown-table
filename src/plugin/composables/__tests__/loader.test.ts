import { describe, it, expect } from 'vitest';
import { ref } from 'vue';
import {
	useIsOnlyLinearLoader,
	useLoaderContainerClasses,
	useLoaderHeight,
	useLoaderTdStyles,
	useLoaderTrStyles,
	useLoaderVRowClasses,
} from '../loader';

describe('Loader Composable', () => {
	describe('useIsOnlyLinearLoader', () => {
		it('should return true if "linear"', () => {
			const data = useIsOnlyLinearLoader('linear');
			expect(data).toBeTruthy();
		});

		it('should return true if "[\'linear\']"', () => {
			const data = useIsOnlyLinearLoader(['linear']);
			expect(data).toBeTruthy();
		});

		it('should return false if "circular"', () => {
			const data = useIsOnlyLinearLoader('circular');
			expect(data).toBeFalsy();
		});

		it('should return false if "[\'linear\', \'circular\']"', () => {
			const data = useIsOnlyLinearLoader(['linear', 'circular']);
			expect(data).toBeFalsy();
		});
	});

	describe('useLoaderContainerClasses', () => {
		it('should return class object using value as true', () => {
			const data = useLoaderContainerClasses({ isLinearOnly: true });
			expect(data).toMatchSnapshot();
		});

		it('should return class object using value as false', () => {
			const data = useLoaderContainerClasses({ isLinearOnly: false });
			expect(data).toMatchSnapshot();
		});

		it('should return class object using value as ref true', () => {
			const data = useLoaderContainerClasses({ isLinearOnly: ref(true) });
			expect(data).toMatchSnapshot();
		});

		it('should return class object using value as ref false', () => {
			const data = useLoaderContainerClasses({ isLinearOnly: ref(false) });
			expect(data).toMatchSnapshot();
		});
	});

	describe('useLoaderHeight', () => {
		it('should return the default of 2px if no value (empty string) is passed in', () => {
			const data = useLoaderHeight('');
			expect(data).toMatchInlineSnapshot(`"2px"`);
		});

		it('should return number value passed in as pixel value', () => {
			const data = useLoaderHeight(5);
			expect(data).toMatchInlineSnapshot(`"5px"`);
		});

		it('should return string value passed in as pixel value', () => {
			const data = useLoaderHeight('5');
			expect(data).toMatchInlineSnapshot(`"5px"`);
		});
	});

	describe('useLoaderTdStyles', () => {
		it('should return styles if isLinearOnly is true and loaderHeight as number', () => {
			const data = useLoaderTdStyles({
				isLinearOnly: true,
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles if isLinearOnly is true and loaderHeight as string', () => {
			const data = useLoaderTdStyles({
				isLinearOnly: true,
				loaderHeight: '5',
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles if isLinearOnly is ref true and loaderHeight as number', () => {
			const data = useLoaderTdStyles({
				isLinearOnly: ref(true),
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return no styles isLinearOnly is false', () => {
			const data = useLoaderTdStyles({
				isLinearOnly: false,
				loaderHeight: 5,
			});
			expect(data).toMatchInlineSnapshot(`{}`);
		});

		it('should return no styles isLinearOnly is ref false', () => {
			const data = useLoaderTdStyles({
				isLinearOnly: ref(false),
				loaderHeight: 5,
			});
			expect(data).toMatchInlineSnapshot(`{}`);
		});
	});

	describe('useLoaderTrStyles', () => {
		it('should return styles if isLinearOnly is true and loaderHeight as number', () => {
			const data = useLoaderTrStyles({
				isLinearOnly: true,
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles if isLinearOnly is true and loaderHeight as string', () => {
			const data = useLoaderTrStyles({
				isLinearOnly: true,
				loaderHeight: '5',
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles if isLinearOnly is ref true and loaderHeight as number', () => {
			const data = useLoaderTrStyles({
				isLinearOnly: ref(true),
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles using 0 as the value if isLinearOnly is false', () => {
			const data = useLoaderTrStyles({
				isLinearOnly: false,
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return styles using 0 as the value if isLinearOnly is ref false', () => {
			const data = useLoaderTrStyles({
				isLinearOnly: ref(false),
				loaderHeight: 5,
			});
			expect(data).toMatchSnapshot();
		});
	});

	describe('useLoaderVRowClasses', () => {
		it('should return classes as object using component name in the vrow class', () => {
			expect(useLoaderVRowClasses()).toMatchSnapshot();
		});
	});
});
