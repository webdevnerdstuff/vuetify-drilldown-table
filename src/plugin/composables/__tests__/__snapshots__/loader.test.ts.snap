// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Composable > useLoaderContainerClasses > should return class object using value as false 1`] = `
{
  "ma-0": true,
  "pa-0": true,
  "text-center": true,
  "v-drilldown-table--loader-tr": true,
  "v-drilldown-table--loader-tr-not-linear": true,
}
`;

exports[`Loader Composable > useLoaderContainerClasses > should return class object using value as ref false 1`] = `
{
  "ma-0": true,
  "pa-0": true,
  "text-center": true,
  "v-drilldown-table--loader-tr": true,
  "v-drilldown-table--loader-tr-not-linear": true,
}
`;

exports[`Loader Composable > useLoaderContainerClasses > should return class object using value as ref true 1`] = `
{
  "ma-0": true,
  "pa-0": true,
  "text-center": true,
  "v-drilldown-table--loader-tr": true,
  "v-drilldown-table--loader-tr-not-linear": false,
}
`;

exports[`Loader Composable > useLoaderContainerClasses > should return class object using value as true 1`] = `
{
  "ma-0": true,
  "pa-0": true,
  "text-center": true,
  "v-drilldown-table--loader-tr": true,
  "v-drilldown-table--loader-tr-not-linear": false,
}
`;

exports[`Loader Composable > useLoaderTdStyles > should return styles if isLinearOnly is ref true and loaderHeight as number 1`] = `
{
  "height": "5px",
  "position": "absolute",
  "top": 0,
  "width": "100%",
}
`;

exports[`Loader Composable > useLoaderTdStyles > should return styles if isLinearOnly is true and loaderHeight as number 1`] = `
{
  "height": "5px",
  "position": "absolute",
  "top": 0,
  "width": "100%",
}
`;

exports[`Loader Composable > useLoaderTdStyles > should return styles if isLinearOnly is true and loaderHeight as string 1`] = `
{
  "height": "5px",
  "position": "absolute",
  "top": 0,
  "width": "100%",
}
`;

exports[`Loader Composable > useLoaderTrStyles > should return styles if isLinearOnly is ref true and loaderHeight as number 1`] = `
{
  "height": "5px",
  "minHeight": "5px",
}
`;

exports[`Loader Composable > useLoaderTrStyles > should return styles if isLinearOnly is true and loaderHeight as number 1`] = `
{
  "height": "5px",
  "minHeight": "5px",
}
`;

exports[`Loader Composable > useLoaderTrStyles > should return styles if isLinearOnly is true and loaderHeight as string 1`] = `
{
  "height": "5px",
  "minHeight": "5px",
}
`;

exports[`Loader Composable > useLoaderTrStyles > should return styles using 0 as the value if isLinearOnly is false 1`] = `
{
  "height": 0,
  "minHeight": 0,
}
`;

exports[`Loader Composable > useLoaderTrStyles > should return styles using 0 as the value if isLinearOnly is ref false 1`] = `
{
  "height": 0,
  "minHeight": 0,
}
`;

exports[`Loader Composable > useLoaderVRowClasses > should return classes as object using component name in the vrow class 1`] = `
{
  "align-center": false,
  "d-grid": false,
  "flex-column": true,
  "ma-0": true,
  "pa-0": true,
  "text-center": true,
  "v-drilldown-table--loader-tr-vrow": true,
}
`;
